### DISCUSSION TITLE
Prettierd help please :)

### DISCUSSION THREAD

**Discussion by AigenUK** [LazyVim/LazyVim#229] [Q&A]
I'm new to neovim and lua scripting. I'm using Lazy.nvim starter and just trying to stop null_ls from updating my code to double quotes and adding semi-colons.
So I've added null_ls.lua file to the plugins directory and using the below code but not working?? Any help really appreciated.

```lua
return {
  "jose-elias-alvarez/null-ls.nvim",
  event = "BufReadPre",
  dependencies = { "mason.nvim" },
  opts = function(_, opts)
    local nls = require("null-ls")
    opts.sources = vim.list_extend(opts.sources, {
      nls.builtins.formatting.prettierd.with({
        extra_args = {
          "--trailing-comma=all",
          "--single-quote=true",
          "--semi=false",
          "--tab-width=2",
          "--print-width=80",
          "--single-attribute-per-line=true",
          "--comma-dangle=always-multiline",
        },
      }),
      nls.builtins.formatting.stylua,
      nls.builtins.diagnostics.flake8,
    })
  end,
}
```

**Comment by folke** [comment]
Can't help you with the null-ls syntax, you have to check the null-ls docs for that.

But for prettier, isn't it easier to have a prettier config file in your project root?
https://prettier.io/docs/en/configuration.html

**Reply by smokeyhallow** [reply]
I too have the same issue, I too prefer to have global config as its the same for all my projects.

**Reply by folke** [reply]
then read how to do that in the null-ls docs as I mentioned? Just **read**

**Comment by leonasdev** [comment]
`prettierd` does not support execution with args.

You can use `prettier` instead to use extra args, but it is generally slower than `prettierd` on large projects.

Alternatively, you can have a `.prettierrc` file under the root of the project or under the env configuration path to default the config.

You can also set the `PRETTIERD_DEFAULT_CONFIG` like this:
```lua
formatting.prettierd.with({
  env = {
    PRETTIERD_DEFAULT_CONFIG = vim.fn.expand("~/.config/.prettierrc.json"),
  },
}),
```


Note if you are using `prettierd`, once you modify `.prettierrc`, you should restart the process of `prettierd` on the system to take effect (which `prettier` no needs).

**Reply by saman-zdf** [reply]
@AigenUK, This config is working for me, just in your null-ls.lus file add it and you should be good to go. Just to let you know, I always have pretteirrc and eslintrc config on all my projects root directory: 
`return {
  "jose-elias-alvarez/null-ls.nvim",
  event = "BufReadPre",
  dependencies = { "mason.nvim" },
  opts = function(_, opts)
    local nls = require("null-ls")
    opts.sources = vim.list_extend(opts.sources, {
      nls.builtins.formatting.prettierd,
      nls.builtins.formatting.stylua,
      nls.builtins.diagnostics.flake8,
    })
  end,
}
`
Also, please double check that your null-ls is actually installed properly. Hope it helps mate.

**Reply by AigenUK** [reply]
Thanks for this, saved me giving up entirely on NeoVim. Appreciate the help and working as expect now... at last :) Thanks again!

**Reply by smokeyhallow** [reply]
@saman-zdf thanks for the comment, but I still can't get it to work. I have reformatted your suggestion and have placed it in `.config/nvim/lua/plugins/null-ls.lua`.

Also, can you explain how the environment variable (`PRETTIERD_DEFAULT_CONFIG`) can be used. I found the [suggestion](https://github.com/jose-elias-alvarez/null-ls.nvim/blob/main/doc/BUILTIN_CONFIG.md#environment-variables) difficult to grasp as to where to place it.


``` lua
return {
  "jose-elias-alvarez/null-ls.nvim",
  event = "BufReadPre",
  dependencies = { "mason.nvim" },
  opts =
      function(_, opts)
        local nls = require("null-ls")
        opts.sources = vim.list_extend(
          opts.sources, {
            nls.builtins.formatting.prettierd,
            nls.builtins.formatting.stylua,
            nls.builtins.diagnostics.flake8,
          })
      end,
}
```

Somehow even *formatting* message is showed (`formatting null-ls`) but nothing happens.

I also found this:

``` bash
~/.config
➜ ps aux | ag prettierd
smokeyhallow    28728  0.0  0.2 592220 42916 ?        Ssl  20:40   0:00 prettierd
smokeyhallow    29112  0.0  0.0   3692  1956 pts/1    S+   20:41   0:00 ag -i prettierd

~/.config
➜ prettierd status
Not running
```

**Reply by folke** [reply]
@smokeyhallow do you have a prettierrc file in your project?

**Reply by folke** [reply]
@smokeyhallow if you want to use a global config, then just use the code with the env that @leonasdev shared.

**Reply by smokeyhallow** [reply]
@folke thanks for answering! So, sorry! It was my mistake! I was working with a *jailed* instance of nvim! That caused prevented the process to talk to `prettierd`.

Its now working!
